### PR TITLE
SAA-600: Return the rollout plan for a prison including rollout status and date. Also refactor DB migrations

### PR DESCRIPTION
**Pull request by smcveigh941**
- Now able to fetch prison rollout information per prison, including a calculated field for the rollout status based on flag and rollout date
- Enabled separate rollout dates for activities and appointments
- Rolled up the baseline DB schemas to prepare for using schema migration scripts

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/RolloutPrison.kt [hunk 1]
```diff
@@ -20,13 +18,25 @@ data class RolloutPrison(
...
 
-  @Enumerated(STRING)
-  private val appointmentsDataSource: AppointmentsDataSource,
+  val appointmentsToBeRolledOut: Boolean,
+
+  val appointmentsRolloutDate: LocalDate? = null,
 ) {
-  fun isAppointmentsEnabled() = this.appointmentsDataSource == AppointmentsDataSource.ACTIVITIES_SERVICE
-  fun isActivitiesEnabled() = this.active
+  fun isActivitiesRolledOut() =
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/RolloutPrison.kt:29]: I think both of these functions can be simplified a little.  There is an extension function available on LocalDate  in the projecte.g.:

```
  fun isActivitiesRolledOut() =
    activitiesToBeRolledOut && activitiesRolloutDate?.onOrBefore(LocalDate.now()) == true

  fun isAppointmentsRolledOut() =
    appointmentsToBeRolledOut && appointmentsRolloutDate?.onOrBefore(LocalDate.now()) == true

```

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/RolloutPrison.kt [hunk 2]
```diff
@@ -20,13 +18,25 @@ data class RolloutPrison(
...
+  val activitiesToBeRolledOut: Boolean,
 
-  val rolloutDate: LocalDate?,
+  val activitiesRolloutDate: LocalDate? = null,
 
-  @Enumerated(STRING)
-  private val appointmentsDataSource: AppointmentsDataSource,
+  val appointmentsToBeRolledOut: Boolean,
+
+  val appointmentsRolloutDate: LocalDate? = null,
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/entity/RolloutPrison.kt:27]: I don't know the answer to this so just asking.  Do we need separate dates for appointments and activities?

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/RolloutController.kt [hunk 3]
```diff
@@ -6,53 +6,73 @@ import io.swagger.v3.oas.annotations.media.Schema
...
+  private val rolloutService: RolloutPrisonService,
 ) {
 
   @GetMapping(value = ["/{prisonCode}"])
-  @ResponseBody
   @Operation(
-    summary = "Get a prison by its code",
-    description = "Returns a single prison and its details by its unique code.",
+    summary = "Get a prison's rollout plan by prison code",
+    description = "Returns a single prison and it's activities management service rollout plan by its unique code.",
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/resource/RolloutController.kt:28]: Minor grammar typo.  Should be **its** and not **it's**.

File: src/main/resources/migrations/dev/V2023.04.27.1__dev_baseline.sql [hunk 4]
```diff
@@ -0,0 +1,52 @@
+-- =============================================
```
Comment (matt-crowson-moj) [inline src/main/resources/migrations/dev/V2023.04.27.1__dev_baseline.sql:1]: I am still on the fence with the dates in the name ...

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/RolloutPrisonPlan.kt [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+  var activitiesRolledOut: Boolean,
+
+  @Schema(description = "The date activities rolled out", example = "2022-09-30")
+  @JsonFormat(pattern = "yyyy-MM-dd")
+  val activitiesRolloutDate: LocalDate?,
+
+  @Schema(description = "Flag to indicate if this prison is presently rolled out for appointments", example = "true")
+  var appointmentsRolledOut: Boolean,
+
+  @Schema(description = "The date activities rolled out", example = "2022-09-30")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/RolloutPrisonPlan.kt:25]: Oops typo or copy and paste ;)  The date appointments rolled out ....

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/RolloutPrisonPlan.kt [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+@Schema(description = "Describes the rollout plan of a prison which may or may not be rolled out")
+data class RolloutPrisonPlan(
+
+  @Schema(description = "The prison code of the requested prison", example = "PVI")
+  var prisonCode: String,
+
+  @Schema(description = "Flag to indicate if this prison is presently rolled out for activities", example = "true")
+  var activitiesRolledOut: Boolean,
+
+  @Schema(description = "The date activities rolled out", example = "2022-09-30")
```
Comment (matt-crowson-moj) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/model/RolloutPrisonPlan.kt:18]: Could say include dates will be null if not rolled out, meh!

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/OffenderDeallocationServiceTest.kt [hunk 7]
```diff
@@ -41,7 +41,7 @@ class OffenderDeallocationServiceTest {
       assertThat(deallocatedBy).isNull()
     }
 
-    whenever(rolloutPrisonRepo.findAllByActiveIsTrue()).thenReturn(listOf(prison))
+    whenever(rolloutPrisonRepo.findAll()).thenReturn(listOf(prison))
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/OffenderDeallocationServiceTest.kt:44]: Should it be `findAll()` here? We'd only want the actively rolled-out prisons?

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceSinglePrisonerTest.kt [hunk 8]
```diff
@@ -267,9 +267,8 @@ class ScheduledEventServiceSinglePrisonerTest {
       setupSinglePrisonerApiMocks(prisonCode, prisonerNumber, dateRangeOverlappingTodaysDate)
 
       setupRolledOutPrisonMock(
-        active = true,
-        rolloutDate = LocalDate.of(2022, 12, 22),
-        AppointmentsDataSource.PRISON_API,
+        activitiesRolloutDate = LocalDate.of(2022, 12, 22),
+        appointmentsRolloutDate = LocalDate.of(2600, 12, 22),
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/ScheduledEventServiceSinglePrisonerTest.kt:271]: I like that the switches are now consistent - it was nagging at me to do something about it.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/InboundEventsServiceTest.kt [hunk 9]
```diff
@@ -21,7 +21,7 @@ class InboundEventsServiceTest {
 
   @Test
   fun `inbound event is processed for active prison`() {
-    repository.stub { on { findByCode(moorlandPrisonCode) } doReturn rolloutPrison().copy(active = true) }
+    repository.stub { on { findByCode(moorlandPrisonCode) } doReturn rolloutPrison() }
```
Comment (timharrison-moj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/service/events/InboundEventsServiceTest.kt:24]: Should only be active, unless you've addressed in the test data by already defaulting to rolledOut == true?

**Review by timharrison-moj — APPROVED** [review]
LGTM - but would you check with Dave please? So its clear what refactoring will be needed because of this? Thanks

File: src/main/resources/migrations/common/V2023.04.27__baseline.sql [hunk 10]
```diff
@@ -0,0 +1,667 @@
...
+insert into activity_tier(activity_tier_id, code, description) values (1, 'Tier1', 'Work, education and maintenance');
+
+alter sequence if exists activity_tier_activity_tier_id_seq restart with 2;
+
+-- =============================================
+-- ATTENDANCE REASONS
+-- =============================================
+
+insert into attendance_reason(attendance_reason_id, code, description, attended, capture_pay,
+                              capture_more_detail, capture_case_note,
```
Comment (timharrison-moj) [inline src/main/resources/migrations/common/V2023.04.27__baseline.sql:621]: Do wonder about combining DDL and base data inserts?

**Review by timharrison-moj — APPROVED** [review]
LGTM

**Pull request merged**